tests/test_container_workflow.py

```
@@ -0,0 +1,424 @@
+        self.test_results = {}
+        self.containers_to_cleanup = []
+    
+    def log(self, message: str, level: str = "INFO"):
+        """Log test messages with timestamp"""
+        timestamp = time.strftime("%H:%M:%S")
+        print(f"[{timestamp}] [{level}] {message}")
+    
+    def run_command(self, cmd: str, capture_output: bool = True, timeout: int = 60) -> Tuple[int, str, str]:
+        """Run shell command and return exit code, stdout, stderr"""
+        try:
+            result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call with `shell=True` identified, security issue

(S602)